plugins/modules/s3_bucket_info.py

```diff
@@ -192,225 +208,311 @@
+                          },
+                          "BucketKeyEnabled": true
+                        }
+                    ]
+                  }
+                }
       contains:
         ServerSideEncryptionConfiguration:
           description: ServerSideEncryptionConfiguration configuration.
-          returned: when encryption is enabled on the bucket
+          returned: when encryption is enabled on the bucket.
           type: complex
           contains:
             Rules:
               description: List of applied encryptio rules.
-              returned: when encryption is enabled on the bucket
+              returned: when encryption is enabled on the bucket.
               type: list
-              sample: { "ApplyServerSideEncryptionByDefault": { "SSEAlgorithm": "AES256" }, "BucketKeyEnabled": False }
+              elements: dict
+              contains:
+                ApplyServerSideEncryptionByDefault:
+                  description: Specifies the default server-side encryption to apply to new objects in the bucket.
+                  type: dict
+                  contains:
+                    SSEAlgorithm:
+                      description: Server-side encryption algorithm to use for the default encryption.
+                      type: str
+                    KMSMasterKeyID:
+                      description: Amazon Web Services Key Management Service (KMS) customer Amazon Web Services KMS key ID to use for the default encryption.
+                      type: str
+                BucketKeyEnabled:
+                  description:
+                    - Specifies whether Amazon S3 should use an S3 Bucket Key with server-side encryption using KMS (SSE-KMS) for new objects in the bucket.
+                  type: bool
     bucket_lifecycle_configuration:
       description: Bucket lifecycle configuration settings.
-      returned: when I(bucket_facts=true) and I(bucket_lifecycle_configuration=true)
-      type: complex
+      returned: when O(bucket_facts=true) and RV(buckets.bucket_lifecycle_configuration=true).
+      type: list
+      elements: dict
       contains:
         Rules:
           description: List of lifecycle management rules.
-          returned: when lifecycle configuration is present
+          returned: when lifecycle configuration is present.
           type: list
-          sample: [{ "Status": "Enabled", "ID": "example-rule" }]
+          sample: [
+                    {
+                      "Status": "Enabled",
+                      "ID": "example-rule"
+                    }
+                  ]
     bucket_location:
       description: Bucket location.
-      returned: when I(bucket_facts=true) and I(bucket_location=true)
-      type: complex
+      returned: when O(bucket_facts=true) and RV(buckets.bucket_location=true).
+      type: dict
+      sample: "bucket_location": {
+                    "LocationConstraint": "us-east-1"
+                }
```

Comment:
```suggestion
      sample: {
                "LocationConstraint": "us-east-1"
              }
```
This should fix the docs/sanity test failure.